src/linear_algebra/matrix.lean

```diff
@@ -936,3 +936,57 @@ def alg_equiv_matrix {R : Type v} {M : Type w} {n : Type*} [fintype n]
   [comm_ring R] [add_comm_group M] [module R M] [decidable_eq n] {b : n → M} (h : is_basis R b) :
   module.End R M ≃ₐ[R] matrix n n R :=
 h.equiv_fun.alg_conj.trans alg_equiv_matrix'
+
+/-- `std_basis_vec` is the standard basis for the free module `n → R`. -/
+def std_basis_vec {R : Type v} [ring R] {n : Type w} [decidable_eq n]
+  (i : n) : n → R := linear_map.std_basis R (λ _, R) i 1
+
+namespace std_basis_vec
+
+open_locale classical
+
+variables {R : Type v} [ring R] {n : Type w}
+
+lemma eq_zero_of_ne {i j : n} (h : i ≠ j) : std_basis_vec i j = (0 : R) :=
+by { rw [std_basis_vec, linear_map.std_basis_ne], exact h.symm }
+
+lemma eq_one_of_eq {i j : n} (h : i = j) : std_basis_vec i j = (1 : R) :=
+by { convert linear_map.std_basis_same _ (λ _, R) j 1 }
+
+@[simp] lemma eq_one {i : n} : std_basis_vec i i = (1 : R) :=
+eq_one_of_eq rfl
+
+lemma is_basis [fintype n] : @is_basis n R (n → R) std_basis_vec _ _ _ :=
+pi.is_basis_fun R n
+
+@[simp] lemma dot_product_eq_val (v : n → R) (i : n) [fintype n]:
+  matrix.dot_product v (std_basis_vec i) = v i :=
```

Comment:
Thoughts on making this just:
```suggestion
  matrix.dot_product v (linear_map.std_basis R (λ _, R) i 1) = v i :=
```
and then just removing all the lemmas above? I know `std_basis` can be awkward to use, but I'm not sure this PR is the right place to introduce an alias for it.

Reply:
Alright, I will use your version